docs/kb/semgrep-ci/jenkins-diff-scans.md

```diff
@@ -0,0 +1,185 @@
+---
+title: Diff-aware scans with GitHub and Jenkins
+description: Set up full and diff-aware scans in Jenkins with Multibranch pipeline projects.
+---
+
+# Full and diff-aware scans with GitHub and Jenkins
+
+This guide provides two setup options: one for full Semgrep scans only, and one for full and diff-aware Semgrep scans. Full scans use a simple Pipeline project, while combining full and diff-aware scans uses a Multibranch Pipeline project. Both options use GitHub as the source code manager, with a repository whose default branch is `main`.
+
+:::info
+Your UI (user interface) may vary depending on your Jenkins installation. These steps use a Classic UI Jenkins interface.
+:::
+
+## Full scans
+
+### Create your Jenkinsfile
+
+To start the process, create your initial `Jenkinsfile` in the root of the repository where you're setting up Semgrep. This code snippet uses Jenkins declarative syntax and runs Semgrep in Docker.
+
+```bash
+pipeline {
+  agent any
+  environment {
+    // Required for a Semgrep AppSec Platform-connected scan:
+    SEMGREP_APP_TOKEN = credentials('SEMGREP_APP_TOKEN')
+  }
+  stages {
+    stage('semgrep-scan') {
+      steps {
+        sh '''docker pull semgrep/semgrep && \
+            docker run \
+            -e SEMGREP_APP_TOKEN=$SEMGREP_APP_TOKEN \
+            -v "$(pwd):$(pwd)" --workdir $(pwd) \
+            semgrep/semgrep semgrep ci '''
+      }
+    }
+  }
+}
+```
+
+This Jenkinsfile uses a `SEMGREP_APP_TOKEN` stored in the Jenkins instance credentials store. It does not set any other variables.
+
+### Set up a pipeline
+
+#### Under General
+
+1. Check the box "GitHub Project" and enter the URL for the project, `https://github.com/<namespace>/<project>/`
+2. Under Build Triggers, select the "GitHub Hook trigger for GITScm polling" box.
+
+#### Under Pipeline
+
```

Comment:
```suggestion
Provide the following details:
```
I'd suggest putting all of the UI elements (e.g., Definition, SCM) in bold and all of the text the user should enter in code font